cmd/config.go

```diff
@@ -0,0 +1,174 @@
+package cmd
+
+import (
+	"fmt"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/srl-labs/containerlab/clab"
+	"github.com/srl-labs/containerlab/clab/config"
+	"github.com/srl-labs/containerlab/types"
+	"golang.org/x/crypto/ssh"
+)
+
+// path to additional templates
+var templatePath string
+
+// Only print config locally, don't send to the node
+var printLines int
+
+// configCmd represents the config command
+var configCmd = &cobra.Command{
+	Use:          "config",
+	Short:        "configure a lab",
+	Long:         "configure a lab based using templates and variables from the topology definition file\nreference: https://containerlab.srlinux.dev/cmd/config/",
+	Aliases:      []string{"conf"},
+	SilenceUsage: true,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var err error
+		if err = topoSet(); err != nil {
+			return err
+		}
+
+		config.DebugCount = debugCount
+
+		opts := []clab.ClabOption{
+			clab.WithDebug(debug),
+			clab.WithTimeout(timeout),
+			clab.WithTopoFile(topo),
+			clab.WithRuntime(rt, debug, timeout, graceful),
+		}
+		c := clab.NewContainerLab(opts...)
+
+		//ctx, cancel := context.WithCancel(context.Background())
+		//defer cancel()
+
+		setFlags(c.Config)
+		log.Debugf("Topology definition: %+v", c.Config)
+		// Parse topology information
+		if err = c.ParseTopology(); err != nil {
+			return err
+		}
+
+		// config map per node. each node gets a couple of config snippets []string
+		allConfig := make(map[string][]config.ConfigSnippet)
+
+		renderErr := 0
+
+		for _, node := range c.Nodes {
+			err = config.LoadTemplate(node.Kind, templatePath)
```

Comment:
shall we make the template load function to use node long name instead? using kind as a map key will limit us in having different templates for different nodes

Reply:
But for the same kind of node you would want the same template? (un-rendered of course)

Maybe you can share an example of your intention?

Reply:
no, I think templates are node-specific, not kind specific
i.e. in my lab with two sros nodes I might want R1 to have a certain template with, say, interfaces + bgp, whereas R2 will have just interfaces template

Do you see this differently?

Reply:
Guess I do see it a bit different. If you want BGP, you will have an additional label/var on that node that either indicates a BGP neighbor address/or the fact that it should peer toward the RR (the RR can be in your template or a global level var) etc. So the level of conditionality is built into the template, since you have to render all templates uniformly over the entire lab, else our command line is broken (we only allow 1 path and one set of template names). 

The example template has `sid_idx` which specifies Segment Routing, or if not there no segment routing if omitted. You can also specify `isis_iid` to change the IGP instance (and in a future commit skip ISIS). This is all part of the IGP part of the `base` template. The lab topology describe all the vars and the templates are responsible to change it into kind specific config. The command line to apply this across the entire lab (could be a mix of kinds) is simply: `clab config -t mylab.clab.yml -p /to/templates -n base`